src/components/WidgetCalendar.vue

```diff
@@ -0,0 +1,156 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+const weeks = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+const todayDate = new Date();
+const currentYear = todayDate.getFullYear();
+const currentMonth = todayDate.getMonth();
+
+const formattedMonthYear = computed(() => {
+  const options: Intl.DateTimeFormatOptions = { month: 'short', year: 'numeric' };
+  return todayDate.toLocaleDateString(undefined, options);
+});
+
+const firstDayOfMonth = new Date(currentYear, currentMonth, 1);
+const lastDayOfMonth = new Date(currentYear, currentMonth + 1, 0);
+
+const lastDateOfPrevMonth = new Date(currentYear, currentMonth, 0).getDate();
+
+const weeksInMonth = computed(() => {
+  const weeksArray: Array<Array<{ date: Date; isToday: boolean; isCurrentMonth: boolean }>> = [];
+  let week: Array<{ date: Date; isToday: boolean; isCurrentMonth: boolean }> = [];
```

Comment:
```suggestion
  const weeksArray: { date: Date; isToday: boolean; isCurrentMonth: boolean }[][] = [];
  let week: { date: Date; isToday: boolean; isCurrentMonth: boolean }[] = [];
```
ついでに`type Day = { date: Date; isToday: boolean; isCurrentMonth: boolean };`とかやってしまってもいいかも